tests/manage/z_cluster/nodes/test_node_replacement_proactive.py

```diff
@@ -104,14 +104,25 @@ def delete_and_create_osd_node(osd_node_name):
             log.error(msg_invalid)
             pytest.fail(msg_invalid)
     elif config.ENV_DATA["platform"].lower() == constants.VSPHERE_PLATFORM:
-        if is_lso_cluster():
-            new_node_name = node.delete_and_create_osd_node_vsphere_upi_lso(
-                osd_node_name, use_existing_node=False
-            )
+        wnodes_not_in_ocs = node.get_worker_nodes_not_in_ocs()
+        # We use for vSphere an existing worker node, due to issue
+        # https://github.com/red-hat-storage/ocs-ci/issues/3894
+        # Issue for tracking is https://github.com/red-hat-storage/ocs-ci/issues/3945
+        if wnodes_not_in_ocs:
```

Comment:
is it always the case that for vsphere, an extra worker non-OCS node has to exist?

Reply:
It was in the past. Then the issue was fixed by Vijay, and we used a new node. We now had this issue again due to a change in the adding node function - We now use terraform for adding nodes.

Reply:
is there any way to first try adding a new node and if it is not possible, look for non OCS worker nodes?

Reply:
Yes. But also in this case, it will not be the same process of the node replacement we want to simulate.